build.xml

```diff
@@ -1,126 +1,417 @@
+
+    <target name="compile-graalSOM" depends="libs" description="Compile GraalSOM">
+        <travis target="compile" start="compile" />
+
         <mkdir dir="${build.dir}"/>
-        <jar destfile="${build.dir}/som.jar" basedir="${classes.dir}">
-        </jar>
+        <mkdir dir="${classes.dir}" />
+        <mkdir dir="${src_gen.dir}" />
+        <javac includeantruntime="false" srcdir="${src.dir}" destdir="${classes.dir}" debug="true">
+            <classpath refid="project.classpath" />
+            <compilerarg line="-s ${src_gen.dir}" />
+            <compilerarg line="-XDignore.symbol.file" />
+            <compilerarg line="-Xlint:all" />
+            <compilerarg line="--release 8" if:true="${is.atLeastJava9}" />
+        </javac>
+        <javac includeantruntime="false" srcdir="${src_gen.dir}" destdir="${classes.dir}" debug="true">
+            <classpath refid="project.classpath" />
+            <compilerarg line="-s ${src_gen.dir}" />
+            <compilerarg line="-Xlint:all" />
+            <compilerarg line="--release 8" if:true="${is.atLeastJava9}" />
+        </javac>
+        <javac includeantruntime="false" srcdir="${unit.dir}" destdir="${classes.dir}" debug="true">
+            <classpath refid="project.classpath" />
+            <compilerarg line="--release 8" if:true="${is.atLeastJava9}" />
+        </javac>
+
+        <travis target="compile" />
+    </target>
+
+    <target name="compile" depends="libs,compile-graalSOM" description="Compile GraalSOM and dependencies">
+    </target>
+
+    <target name="jar" depends="compile" description="Package as JAR">
+        <jar destfile="${build.dir}/graal-som.jar" basedir="${classes.dir}"></jar>
     </target>
 
     <target name="unit-tests" depends="compile" description="Execute tests">
         <junit haltonerror="false" haltonfailure="false" failureproperty="test.failed"
-            outputtoformatters="true">
+               outputtoformatters="true">
             <jvmarg value="-ea" />
             <jvmarg value="-esa" />
             <classpath refid="project.classpath" />
             <batchtest fork="yes" filtertrace="false">
-              <fileset dir="tests">
-                  <include name="**/*Test*.java"/>
-                  <include name="**/*TCK.java"/>
-                  <exclude name="dym"/>
-              </fileset>
+            <fileset dir="tests">
+                    <include name="**/*Test*.java"/>
+                </fileset>
             </batchtest>
             <formatter type="brief" usefile="false"/>
         </junit>
 
-        <java classname="som.vm.Universe" fork="true" failonerror="true">
+        <java classname="som.Launcher" fork="true" failonerror="true">
             <classpath refid="project.classpath" />
             <jvmarg value="-ea" />
             <jvmarg value="-esa" />
-            <arg line="-cp Smalltalk TestSuite/TestHarness.som" />
+            <arg line="-cp Smalltalk TestSuite/TestHarness.som  --ignore-inefficacies" />
```

Comment:
Hm, is this necessary? Do you have things that fail without it?

Reply:
It does not fail without it, and I don't even remember why I put this here in the first place. I removed it.